datafusion/expr/src/type_coercion/functions.rs

```diff
@@ -104,23 +105,28 @@ fn get_valid_types(
         let elem_base_type = datafusion_common::utils::base_type(elem_type);
         let new_base_type = comparison_coercion(&array_base_type, &elem_base_type);
 
-        if new_base_type.is_none() {
-            return internal_err!(
+        let new_base_type = new_base_type.ok_or_else(|| {
+            internal_datafusion_err!(
                 "Coercion from {array_base_type:?} to {elem_base_type:?} not supported."
-            );
-        }
-        let new_base_type = new_base_type.unwrap();
+            )
+        })?;
 
         let array_type = datafusion_common::utils::coerced_type_with_base_type_only(
             array_type,
             &new_base_type,
         );
 
         match array_type {
-            DataType::List(ref field) | DataType::LargeList(ref field) => {
-                let elem_type = field.data_type();
+            DataType::List(ref field)
+            | DataType::LargeList(ref field)
+            | DataType::FixedSizeList(ref field, _) => {
+                let elem_type = if elem_base_type.eq(&DataType::Null) {
+                    field.data_type()
+                } else {
+                    elem_type
```

Comment:
why is `let elem_type = field.data_type();` incorrect?

Reply:
Oh, it seems that `array_and_element` 's element type is i64.

Reply:
Will it clear that `array_and_elem` has elem_type as `field.data_type()`, and `array_and_index` has index_type that does not need coercion.